File: 1544-make-the-string-great/1544-make-the-string-great.java
class Solution {
    public String makeGood(String s) {
         int n = s.length();
        
        for (int i = 0; i < n - 1; i++) {
            if (Math.abs(s.charAt(i) - s.charAt(i+1)) == 32) {
                return makeGood(s.substring(0, i) + s.substring(i+2));
            }
        }
        
        return s;
    }
}

File: 1544-make-the-string-great/1544-make-the-string-great.cpp
class Solution {
public:
    string makeGood(string s) {
      int n = s.size();
        
        for (int i = 0; i < n - 1; i++) {
            if (abs(s[i] - s[i+1]) == 32) {
                return makeGood(s.substr(0, i) + s.substr(i+2));
            }
        }
        
        return s;
        
    }
};